parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -210,23 +210,31 @@ configureCNIIPTables() {
     fi
 }
 
+disableSystemdResolved() {
+    echo "Ignoring systemd-resolved query service but using its resolv.conf file"
+    echo "This is the simplest approach to workaround resolved issues without completely uninstalling it"
+    [ -f /run/systemd/resolve/resolv.conf ] && sudo ln -sf /run/systemd/resolve/resolv.conf /etc/resolv.conf
+    ls -ltr /etc/resolv.conf
+    cat /etc/resolv.conf
+}
+
 disable1804SystemdResolved() {
     ls -ltr /etc/resolv.conf
     cat /etc/resolv.conf
     {{- if Disable1804SystemdResolved}}
     UBUNTU_RELEASE=$(lsb_release -r -s)
     if [[ ${UBUNTU_RELEASE} == "18.04" ]]; then
-        echo "Ingorings systemd-resolved query service but using its resolv.conf file"
-        echo "This is the simplest approach to workaround resolved issues without completely uninstall it"
-        [ -f /run/systemd/resolve/resolv.conf ] && sudo ln -sf /run/systemd/resolve/resolv.conf /etc/resolv.conf
-        ls -ltr /etc/resolv.conf
-        cat /etc/resolv.conf
+        disableSystemdResolved
     fi
     {{- else}}
     echo "Disable1804SystemdResolved is false. Skipping."
     {{- end}}
 }
 
+disableSystemdIptables() {
+    systemctlDisableAndStop iptables
```

Comment:
Oh interesting, Mariner distros have some type of iptables service managed by systemd? Do you know what it's doing exactly?
Disabling this systemd service shouldn't affect kube-proxy and CNI functionality right?

Reply:
The systemd iptables service in Mariner doesn't do much actively, it just runs a script on start and stop that configure the iptables rules. By default this is a really strict config, the service [sets](https://github.com/microsoft/CBL-Mariner/blob/ee38a79ca85af4a7eafc2e0921188f4ff6f0db9d/SPECS/iptables/iptables#L74) the [filter table's chain policies to DROP](https://github.com/microsoft/CBL-Mariner/blob/ee38a79ca85af4a7eafc2e0921188f4ff6f0db9d/SPECS/iptables/ip4save#L14).

While K8s is able to set its rules in the chains whether or not this service is running, it's not expecting the chain policy to be DROP instead of ACCEPT. By [stopping the service](https://github.com/microsoft/CBL-Mariner/blob/ee38a79ca85af4a7eafc2e0921188f4ff6f0db9d/SPECS/iptables/iptables.stop#L12), or never starting it on a boot, the chains default to ACCEPT, which matches other AKS distros.

With the systemd iptables service disabled it looks like kube-proxy and CNI are working just fine.